pkg/conversions/virtualrouter_types_conversion.go

```diff
@@ -0,0 +1,327 @@
+		}
+	}
+
+	sdkObj.Metadata = sdkMetadataList
+	sdkObj.MethodName = crdObj.MethodName
+	sdkObj.ServiceName = crdObj.ServiceName
+	return nil
+
+}
+
+func Convert_CRD_GRPCRouteAction_To_SDK_GrpcRouteAction(crdObj *appmesh.GRPCRouteAction,
+	sdkObj *appmeshsdk.GrpcRouteAction, scope conversion.Scope) error {
+
+	var sdkWeightedTargets []*appmeshsdk.WeightedTarget
+	if len(crdObj.WeightedTargets) != 0 {
+		sdkWeightedTargets = make([]*appmeshsdk.WeightedTarget, 0, len(crdObj.WeightedTargets))
+		for _, crdWeightedTarget := range crdObj.WeightedTargets {
+			sdkWeightedTarget := &appmeshsdk.WeightedTarget{}
+			if err := Convert_CRD_WeightedTarget_To_SDK_WeightedTarget(&crdWeightedTarget, sdkWeightedTarget, scope); err != nil {
+				return err
+			}
+			sdkWeightedTargets = append(sdkWeightedTargets, sdkWeightedTarget)
+		}
+	}
+	sdkObj.WeightedTargets = sdkWeightedTargets
+	return nil
+}
+
+func Convert_CRD_GRPCRetryPolicy_To_SDK_GrpcRetryPolicy(crdObj *appmesh.GRPCRetryPolicy,
+	sdkObj *appmeshsdk.GrpcRetryPolicy, scope conversion.Scope) error {
+
+	var sdkGrpcRetryEvents []*string
+	if len(crdObj.GRPCRetryEvents) != 0 {
+		sdkGrpcRetryEvents = make([]*string, 0, len(crdObj.GRPCRetryEvents))
+		for _, crdGRPCRetryEvent := range crdObj.GRPCRetryEvents {
+			sdkGrpcRetryEvents = append(sdkGrpcRetryEvents, aws.String((string)(crdGRPCRetryEvent)))
+		}
+	}
+	sdkObj.GrpcRetryEvents = sdkGrpcRetryEvents
+
+	var sdkHttpRetryEvents []*string
+	if len(crdObj.HTTPRetryEvents) != 0 {
+		sdkHttpRetryEvents = make([]*string, 0, len(crdObj.HTTPRetryEvents))
+		for _, crdHTTPRetryEvent := range crdObj.HTTPRetryEvents {
+			sdkHttpRetryEvents = append(sdkHttpRetryEvents, aws.String((string)(crdHTTPRetryEvent)))
+		}
+	}
+	sdkObj.HttpRetryEvents = sdkHttpRetryEvents
+
+	var sdkTcpRetryEvents []*string
+	if len(crdObj.TCPRetryEvents) != 0 {
+		sdkTcpRetryEvents = make([]*string, 0, len(crdObj.TCPRetryEvents))
+		for _, crdTCPRetryEvent := range crdObj.TCPRetryEvents {
+			sdkTcpRetryEvents = append(sdkTcpRetryEvents, aws.String((string)(crdTCPRetryEvent)))
+		}
+	}
+	sdkObj.TcpRetryEvents = sdkTcpRetryEvents
+
+	sdkObj.PerRetryTimeout = &appmeshsdk.Duration{}
+	if err := Convert_CRD_Duration_To_SDK_Duration(&crdObj.PerRetryTimeout, sdkObj.PerRetryTimeout, scope); err != nil {
+		return err
+	}
+
+	sdkObj.MaxRetries = aws.Int64((int64)(crdObj.MaxRetries))
+	return nil
+}
+
+func Convert_CRD_GRPCRoute_To_SDK_GrpcRoute(crdObj *appmesh.GRPCRoute,
+	sdkObj *appmeshsdk.GrpcRoute, scope conversion.Scope) error {
+
+	sdkObj.Match = &appmeshsdk.GrpcRouteMatch{}
+	if err := Convert_CRD_GRPCRouteMatch_To_SDK_GrpcRouteMatch(&crdObj.Match, sdkObj.Match, scope); err != nil {
+		return err
+	}
+
+	sdkObj.Action = &appmeshsdk.GrpcRouteAction{}
+	if err := Convert_CRD_GRPCRouteAction_To_SDK_GrpcRouteAction(&crdObj.Action, sdkObj.Action, scope); err != nil {
+		return err
+	}
+
+	sdkObj.RetryPolicy = &appmeshsdk.GrpcRetryPolicy{}
```

Comment:
we need to check crdObj.RetryPolicy nil here

Reply:
Done